### PR TITLE
Sewerwater chart redesign

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard1**
## Summary

Before the sewer water line chart only showed the weekly averages. This has been redesigned so that it also shows the individual RWZI data as a scatter plot in the background. An optional sewer station selector was added that allows the data of a specific station to be rendered as an extra line in the chart.
Lastly, the time controls were added as well so that a filter for all, last 5 weeks or last week can be set.

## Motivation

Feature request

## Detailed design

A new component RegionalSewerWaterChart was added that receives all of its data and filter settings through its props.
A hook called useRegionalSewerWaterChartOptions was added that takes all of this data and shapes it into a valid Highcharts configuration which is fed to a generic React Highcharts component.
After another fight with the Highcharts configuration it was decided to render the legend manually instead of using the default Highcharts version.

## Drawbacks

n/a

## Alternatives

n/a

## Unresolved questions

n/a

### Governance

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

File: src/utils/sewer-water/safety-region-sewer-water.util.ts [hunk 1]
```diff
@@ -94,6 +94,33 @@ export function getSewerWaterBarScaleData(
...
 
+export function getSewerWaterStationNames(data?: Regionaal): string[] {
+  const { dataAvailable, oneInstallation } = getSewerWaterMetadata(data);
+
+  if (!data || !dataAvailable || oneInstallation) {
+    return [];
+  }
+
+  return data.results_per_sewer_installation_per_region.values
+    .flatMap((value) => value.values)
```
Comment (VWSCoronaDashboard4) [inline src/utils/sewer-water/safety-region-sewer-water.util.ts:105]: flatmap does not work in IE, do we have a polyfill for this?

File: src/utils/sewer-water/safety-region-sewer-water.util.ts [hunk 2]
```diff
@@ -94,6 +94,33 @@ export function getSewerWaterBarScaleData(
   }
 }
 
+export function getSewerWaterStationNames(data?: Regionaal): string[] {
```
Comment (VWSCoronaDashboard4) [inline src/utils/sewer-water/safety-region-sewer-water.util.ts:97]: `getInstallationNames`

File: src/components/lineChart/rwziselector.module.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  padding: 0.5em;
+  padding-right: 1.5em;
+  text-align: left;
+  background-color: white;
+  background: url('/images/close.svg');
+  background-repeat: no-repeat, repeat;
+  background-position: right 0.2em top 50%, 0 0;
+}
+
+.rwziSelector {
```
Comment (VWSCoronaDashboard4) [inline src/components/lineChart/rwziselector.module.scss:19]: `.installationSelector`

File: src/pages/veiligheidsregio/[code]/rioolwater.tsx [hunk 4]
```diff
@@ -23,20 +24,38 @@ import {
...
       barScaleData: getSewerWaterBarScaleData(data),
       lineChartData: getSewerWaterLineChartData(data),
       barChartData: getSewerWaterBarChartData(data),
+      scatterPlotData: getSewerWaterScatterPlotData(data),
+      sewerStationNames: getSewerWaterStationNames(data),
     };
   }, [data]);
 
+  const [timeframe, setTimeframe] = useState<TimeframeOption>('all');
+  const [selectedRWZI, setSelectedRWZI] = useState<string | undefined>();
```
Comment (VWSCoronaDashboard4) [inline src/pages/veiligheidsregio/[code]/rioolwater.tsx:57]: `[selectedInstallation, setSelectedInstallation]`

File: src/scss/layout.scss [hunk 5]
```diff
@@ -74,6 +74,24 @@
   }
 }
 
+.metric-article-header {
+  display: flex;
+  flex-direction: row;
+  align-items: flex-start;
+
+  div {
```
Comment (VWSCoronaDashboard4) [inline src/scss/layout.scss:82]: This is perhaps due to the way these headers are structured; but I'd use a classname here to be more specific than just `div`

**Review by VWSCoronaDashboard4 — COMMENTED** [review]
Commented some nitpicks and opinions. I'd name all the uses where now the variable name / classname / etc "rwzi" or "sewerwaterstation" is used simply "installation".

File: src/utils/sewer-water/safety-region-sewer-water.util.ts (hunk 1 above)
Comment (VWSCoronaDashboard1) [reply src/utils/sewer-water/safety-region-sewer-water.util.ts:105]: Its being used in other parts of the code base already, so I assumed it is polyfilled already.

File: src/scss/layout.scss (hunk 5 above)
Comment (VWSCoronaDashboard1) [reply src/scss/layout.scss:82]: I unfortunately don't have control over the classname of that div, this is a bit of a hack to style the ChartTimeControls div

File: src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+        },
+      },
+      title: {
+        text: undefined,
+      },
+      tooltip: {
+        backgroundColor: '#FFF',
+        borderColor: '#01689B',
+        borderRadius: 0,
+        formatter: function (): false | string {
```
Comment (VWSCoronaDashboard8) [inline src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts:185]: nit; Return type definition should be inferred from function implementation

**Review by VWSCoronaDashboard8 — APPROVED** [review]

**Pull request merged**